pkgs/development/python-modules/opensfm/default.nix

```diff
@@ -0,0 +1,118 @@
+{ lib
+, fetchurl
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, opencv
+, opencv4
+, ceres-solver
+, suitesparse
+, metis
+, eigen
+, pkg-config
+, pybind11
+, numpy
+, pyyaml
+, lapack
+, gtest
+, gflags
+, glog
+, pytestCheckHook
+, networkx
+, pillow
+, exifread
+, gpxpy
+, pyproj
+, python-dateutil
+, joblib
+, repoze_lru
+, xmltodict
+, cloudpickle
+, scipy
+, sphinx
+, matplotlib
+, fpdf
+}:
+
+let
+  pendingPRs.fixReports = fetchurl {
+    url = "https://github.com/mapillary/OpenSfM/commit/7c4477c6416417a935dd8cfb59e6e7e60cc2ba37.patch";
+    sha256 = "sha256-i6p/K2vIKouY8Fmp9iKdc7Xk2u7o51PQT5fhnKzqmhs=";
+  };
+in buildPythonPackage rec {
+  pname = "OpenSfM";
+  version = "0.5.2";
+  src = fetchFromGitHub {
+    owner = "mapillary";
+    repo = "OpenSfM";
+    # Fixing by commit-id until the upstream issues are solved
+    rev = "2a1c9be07f47f6e26f87fedf8275fcf8ff7b8487";
+    sha256 = "sha256-CARPFuYypgV1YcqZp//Qws8YrV25Z+w/dWB7MbFAKys=";
+    fetchSubmodules = true;
```

Comment:
Why do we use their pybind11? Can't we use the nixpkgs one?

Reply:
I added the changes necessary to use nixpkgs' pybind, but I'm considering reverting them: they used an older version of pybind11 and had their own versions of utilities that have been later introduced upstream (ie in the new pybind11). These are inconsistent